src/poisson_dev_fe.jl

```diff
@@ -571,6 +571,29 @@ low_level_manual_gradient_dv_array = lazy_map(Broadcasting(Operation(⋅)),inv_J
 
 # ## A low-level implementation of the residual integration and assembly
 
+# The discretized variational form of the Poisson problem reads as:
```

Comment:
I would start here with some introductory text towards an smoother transition + some motivation. Something like (did not check that renders ok, might need edits, etc.):

```
In the rest of the tutorial we aim to solve a Poisson equation with homogeneous source term, i.e.,  `$f=0$` 
and non-homogeneous Dirichlet boundary conditions $u=g_0$ on $\Gamma_D$, with  $\Gamma_D$ being
 the whole boundary of the box. While the strong imposition of non-homogeneous Dirichlet boundary 
conditions in Gridap is done under the hood by modifying the global assembly process by subtracting 
the contributions of boundary conditions to the right hand side of the linear system, in this tutorial,
 for simplicity, we follow a different approach. This approach requires: (1) to be able to assemble the 
residual of the PDE for an arbitrary finite element function $\hat{u}_h$ (current section). (2) to be able 
to assemble the coefficient matrix of the finite element linear system (next section). We briefly outline 
this solution approach in the sequel. 
```
What do you think?

Reply:
The background introduction is definitely needed.